cpp/cppsdk/gsdk.cpp

```diff
@@ -50,6 +50,7 @@ namespace Microsoft
                     config = configSmrtPtr.get();
                 }
 
+				m_configMutex.lock();
```

Comment:
We don't need to lock in the constructor, because again, no one can access it until the constructor has completed.